[CI] CsvIT test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping} failing

**Build Scans:**
- [elasticsearch-intake #35589 / part3](https://gradle-enterprise.elastic.co/s/dpgwlb4gsknxq)
- [elasticsearch-periodic-java-ea #197 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/w4qps4qc2dmzq)
- [elasticsearch-pull-request #126128 / part-3](https://gradle-enterprise.elastic.co/s/vqvj7makuw4nk)
- [elasticsearch-pull-request #125948 / part-3](https://gradle-enterprise.elastic.co/s/o2jzgg4icdm5q)
- [elasticsearch-pull-request #125859 / part-3](https://gradle-enterprise.elastic.co/s/j2jpaqflwcprk)
- [elasticsearch-pull-request #125305 / part-3](https://gradle-enterprise.elastic.co/s/tpal5srryllhu)
- [elasticsearch-pull-request #124979 / part-3](https://gradle-enterprise.elastic.co/s/mgjwyxxwl3oo6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.CsvIT" -Dtests.method="test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping}" -Dtests.seed=B5FA5876E4E83BC8 -Dtests.locale=pa-IN -Dtests.timezone=America/Kralendijk -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.CsvIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Ak8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 6 column 0:a list containing
row 6 column 0:0: expected <39.54166666666667> but was <39.541666666666664>
Actual:
cost:double        | time_bucket:datetime
69.6875            | 2024-05-10T00:09:00.000Z
56.5625            | 2024-05-10T00:08:00.000Z
49.5               | 2024-05-10T00:17:00.000Z
48.3125            | 2024-05-10T00:22:00.000Z
45.8125            | 2024-05-10T00:15:00.000Z
40.4375            | 2024-05-10T00:06:00.000Z
39.541666666666664 | 2024-05-10T00:13:00.000Z
37.9375            | 2024-05-10T00:12:00.000Z
36.6875            | 2024-05-10T00:19:00.000Z
36.375             | 2024-05-10T00:11:00.000Z

Expected:
cost:double       | time_bucket:datetime
69.6875           | 2024-05-10T00:09:00.000Z
56.5625           | 2024-05-10T00:08:00.000Z
49.5              | 2024-05-10T00:17:00.000Z
48.3125           | 2024-05-10T00:22:00.000Z
45.8125           | 2024-05-10T00:15:00.000Z
40.4375           | 2024-05-10T00:06:00.000Z
39.54166666666667
[truncated]
```

**Issue Reasons:**
- [main] 7 failures in test test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping} (0.8% fail rate in 886 executions)
- [main] 5 failures in step part-3 (0.8% fail rate in 595 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (0.9% fail rate in 587 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -372,30 +372,6 @@ tests:
 - class: org.elasticsearch.index.query.PrefixQueryBuilderTests
   method: testPrefixCircuitBreakerTripsWithLowLimit
   issue: https://github.com/elastic/elasticsearch/issues/143548
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-irate.Irate_of_ratio_promql}
-  issue: https://github.com/elastic/elasticsearch/issues/143616
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-irate.Irate_of_ratio}
-  issue: https://github.com/elastic/elasticsearch/issues/143617
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-irate.Irate_of_double_no_grouping_promql}
-  issue: https://github.com/elastic/elasticsearch/issues/143624
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-irate.Irate_of_double_no_grouping}
-  issue: https://github.com/elastic/elasticsearch/issues/143625
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_aggregate_metric_double}
-  issue: https://github.com/elastic/elasticsearch/issues/143631
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-delta.Delta_of_double_no_grouping}
-  issue: https://github.com/elastic/elasticsearch/issues/143632
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_aggregate_metric_double_grouping}
-  issue: https://github.com/elastic/elasticsearch/issues/143633
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-delta.Delta_of_double_no_grouping_promql}
-  issue: https://github.com/elastic/elasticsearch/issues/143634
 - class: org.elasticsearch.search.SearchLoggingIT
   method: testSearchLogShardInfoPartialFailure
   issue: https://github.com/elastic/elasticsearch/issues/143638
@@ -423,12 +399,6 @@ tests:
 - class: org.elasticsearch.compute.lucene.query.LuceneTopNSourceOperatorScoringTests
   method: testAccumulateSearchLoad
   issue: https://github.com/elastic/elasticsearch/issues/143750
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping}
-  issue: https://github.com/elastic/elasticsearch/issues/143757
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:k8s-timeseries-avg-over-time.Avg_over_time_of_double_no_grouping_promql}
-  issue: https://github.com/elastic/elasticsearch/issues/143758
 - class: org.elasticsearch.xpack.esql.action.EsqlQueryLoggingIT
   method: testLogging
   issue: https://github.com/elastic/elasticsearch/issues/143780

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/k8s-timeseries-avg-over-time.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/k8s-timeseries-avg-over-time.csv-spec
@@ -2,7 +2,10 @@ avg_over_time_of_double_no_grouping
 required_capability: ts_command_v0
 TS k8s
 | STATS cost=sum(avg_over_time(network.cost)) BY time_bucket = bucket(@timestamp,1minute)
-| SORT cost DESC, time_bucket DESC | LIMIT 10;
+| EVAL cost = round(cost, 4)
+| KEEP cost, time_bucket
+| SORT cost DESC, time_bucket DESC 
+| LIMIT 10;
 
 cost:double       | time_bucket:datetime
 69.6875           | 2024-05-10T00:09:00.000Z
@@ -11,7 +14,7 @@ cost:double       | time_bucket:datetime
 48.3125           | 2024-05-10T00:22:00.000Z
 45.8125           | 2024-05-10T00:15:00.000Z
 40.4375           | 2024-05-10T00:06:00.000Z
-39.54166666666667 | 2024-05-10T00:13:00.000Z
+39.5417           | 2024-05-10T00:13:00.000Z
 37.9375           | 2024-05-10T00:12:00.000Z
 36.6875           | 2024-05-10T00:19:00.000Z
 36.375            | 2024-05-10T00:11:00.000Z
@@ -20,7 +23,10 @@ cost:double       | time_bucket:datetime
 avg_over_time_of_double_no_grouping_promql
 required_capability: promql_command_v0
 PROMQL index=k8s step=1m cost=(sum(avg_over_time(network.cost[1m])))
-| SORT cost DESC, step DESC | LIMIT 10;
+| EVAL cost = round(cost, 4)
+| KEEP cost, step
+| SORT cost DESC, step DESC 
+| LIMIT 10;
 
 cost:double       | step:datetime
 69.6875           | 2024-05-10T00:09:00.000Z
@@ -29,7 +35,7 @@ cost:double       | step:datetime
 48.3125           | 2024-05-10T00:22:00.000Z
 45.8125           | 2024-05-10T00:15:00.000Z
 40.4375           | 2024-05-10T00:06:00.000Z
-39.54166666666667 | 2024-05-10T00:13:00.000Z
+39.5417           | 2024-05-10T00:13:00.000Z
 37.9375           | 2024-05-10T00:12:00.000Z
 36.6875           | 2024-05-10T00:19:00.000Z
 36.375            | 2024-05-10T00:11:00.000Z

